devito/types/sparse.py

```diff
@@ -585,9 +591,9 @@ def gridpoints(self):
             raise ValueError("No coordinates attached to this SparseFunction")
         ret = []
         for coords in self.coordinates.data._local:
-            ret.append(tuple(int(sympy.floor((c - o.data)/i.spacing.data)) for c, o, i in
+            ret.append(tuple(int(np.floor(c - o.data)/i.spacing.data) for c, o, i in
```

Comment:
@fablup this `sympy.floor` was super bad actually

Reply:
OK. Worth trying it out again w/o memoization?

Reply:
Did, still need it for `dist_map` but removed the mempization on `_support` that is now way faster (most time was spent here)

Reply:
question: after changing from sympy.floor to np.floor, what is the next hotspot ?

Reply:
`glb_to_rank`  is where most of the time is spent now